### ISSUE TITLE
wrong results in term matching

### ISSUE THREAD

**Issue by MichaelJCompton** [dgraph-io/dgraph#1001]
On play.dgraph.io run

```
{
  me(func: allofterms(name@en, "neve")) {
    name@en
  }
}
```

Some of the results are "Buddy Goes West", "The Name of the Rose" and "Aliens".

Same goes for anyofterms, but not FTS.

**Comment by MichaelJCompton** [comment]
Another example.

I'm looking for the movie 'Eat Drink Man Woman'

```
{
	edmw(func: allofterms(name@en, "Eat Drink Man Woman")) {
    name@en
  }
}
```
returns no results

but
```
{
	edmw(func: allofterms(name@en, "Eat Drink")) {
    name@en
  }
}
```
finds the movie

while
```
{
	edmw(func: allofterms(name@en, "Man Woman")) {
    name@en
  }
}
```
finds other things with 'man' and 'woman', but not this movie.

**Comment by pawanrawal** [comment]
This was caused by a bug in the filtering logic for uids when predicate had a lang tag. The fix was has been merged to `release/v0.7.7` and `master`.
